src/Categories/Bicategory/Construction/Bimodules/Tensorproduct.agda

```diff
@@ -0,0 +1,727 @@
+        actionʳ₂ ◁ F₁ ∘ᵥ (η₃ ◁ F₂ ∘ᵥ unitorˡ.to) ◁ F₁ ≈⟨ ∘ᵥ-distr-◁ ⟩
+        (actionʳ₂ ∘ᵥ η₃ ◁ F₂ ∘ᵥ unitorˡ.to) ◁ F₁ ≈⟨ ◁-resp-≈ identityʳ₂ ⟩
+        id₂ ◁ F₁ ≈⟨ id₂◁ ⟩
+        id₂ ∎
+        where
+          open hom.HomReasoning
+
+      identityʳ∘arr : (actionʳ ∘ᵥ η₃ ◁ F ∘ᵥ unitorˡ.to) ∘ᵥ Coequalizer.arr F₂⊗F₁ ≈ id₂ ∘ᵥ Coequalizer.arr F₂⊗F₁
+      identityʳ∘arr = begin
+        (actionʳ ∘ᵥ η₃ ◁ F ∘ᵥ unitorˡ.to) ∘ᵥ Coequalizer.arr F₂⊗F₁ ≈⟨ assoc₂ ⟩
+        actionʳ ∘ᵥ (η₃ ◁ F ∘ᵥ unitorˡ.to) ∘ᵥ Coequalizer.arr F₂⊗F₁ ≈⟨ refl⟩∘⟨ assoc₂ ⟩
+        actionʳ ∘ᵥ η₃ ◁ F ∘ᵥ unitorˡ.to ∘ᵥ Coequalizer.arr F₂⊗F₁ ≈⟨ refl⟩∘⟨ refl⟩∘⟨ ⟺ ▷-∘ᵥ-λ⇐ ⟩
+        actionʳ ∘ᵥ η₃ ◁ F ∘ᵥ id₁ ▷ Coequalizer.arr F₂⊗F₁ ∘ᵥ unitorˡ.to ≈⟨ refl⟩∘⟨ sym-assoc₂ ⟩
+        actionʳ ∘ᵥ (η₃ ◁ F ∘ᵥ id₁ ▷ Coequalizer.arr F₂⊗F₁) ∘ᵥ unitorˡ.to ≈⟨ refl⟩∘⟨ ⟺ ◁-▷-exchg ⟩∘⟨refl ⟩
+        actionʳ ∘ᵥ (T₃ ▷ Coequalizer.arr F₂⊗F₁ ∘ᵥ η₃ ◁ (F₂ ∘₁ F₁)) ∘ᵥ unitorˡ.to ≈⟨ refl⟩∘⟨ assoc₂ ⟩
+        actionʳ ∘ᵥ T₃ ▷ Coequalizer.arr F₂⊗F₁ ∘ᵥ η₃ ◁ (F₂ ∘₁ F₁) ∘ᵥ unitorˡ.to ≈⟨ sym-assoc₂ ⟩
+        (actionʳ ∘ᵥ T₃ ▷ Coequalizer.arr F₂⊗F₁) ∘ᵥ η₃ ◁ (F₂ ∘₁ F₁) ∘ᵥ unitorˡ.to ≈⟨ ⟺ actionʳSq ⟩∘⟨refl ⟩
+        (Coequalizer.arr F₂⊗F₁ ∘ᵥ actionʳ₂◁F₁) ∘ᵥ η₃ ◁ (F₂ ∘₁ F₁) ∘ᵥ unitorˡ.to ≈⟨ assoc₂ ⟩
+        Coequalizer.arr F₂⊗F₁ ∘ᵥ actionʳ₂◁F₁ ∘ᵥ η₃ ◁ (F₂ ∘₁ F₁) ∘ᵥ unitorˡ.to ≈⟨ refl⟩∘⟨ identityʳ-triangle ⟩
+        Coequalizer.arr F₂⊗F₁ ∘ᵥ id₂ ≈⟨ identity₂ʳ ⟩
+        Coequalizer.arr F₂⊗F₁ ≈⟨ ⟺ identity₂ˡ ⟩
+        id₂ ∘ᵥ Coequalizer.arr F₂⊗F₁ ∎
+        where
+          open hom.HomReasoning
+
+      identityʳ : actionʳ ∘ᵥ (η₃ ◁ F) ∘ᵥ unitorˡ.to ≈ id₂
+      identityʳ = Coequalizer⇒Epi (hom C₁ C₃) F₂⊗F₁ (actionʳ ∘ᵥ (η₃ ◁ F) ∘ᵥ unitorˡ.to) id₂ identityʳ∘arr
+    -- end abstract --
+
+  B₂⊗B₁ : Bimodule M₁ M₃
+  B₂⊗B₁ = record
+    { F = F
+    ; actionˡ = Left-Action.actionˡ --: F ∘₁ T₁ ⇒₂ F  
+    ; actionʳ = Right-Action.actionʳ --: T₂ ∘₁ F ⇒₂ F 
+    ; assoc = Associativity.assoc    -- : actionʳ ∘ᵥ (T₂ ▷ actionˡ) ∘ᵥ associator.from ≈ actionˡ ∘ᵥ (actionʳ ◁ T₁)
+    ; sym-assoc = Associativity.sym-assoc --: actionˡ ∘ᵥ (actionʳ ◁ T₁)∘ᵥ associator.to ≈ actionʳ ∘ᵥ (T₂ ▷ actionˡ)
+    ; assoc-actionˡ = Associativity.assoc-actionˡ     --: actionˡ ∘ᵥ (F ▷ μ₁) ∘ᵥ associator.from ≈ actionˡ ∘ᵥ (actionˡ ◁ T₁)
+    ; sym-assoc-actionˡ = Associativity.sym-assoc-actionˡ --: actionˡ ∘ᵥ (actionˡ ◁ T₁) ∘ᵥ associator.to ≈ actionˡ ∘ᵥ (F ▷ μ₁)
+    ; assoc-actionʳ = Associativity.assoc-actionʳ     --: actionʳ ∘ᵥ (μ₂ ◁ F) ∘ᵥ associator.to ≈ actionʳ ∘ᵥ (T₂ ▷ actionʳ)
+    ; sym-assoc-actionʳ = Associativity.sym-assoc-actionʳ --: actionʳ ∘ᵥ (T₂ ▷ actionʳ) ∘ᵥ associator.from ≈ actionʳ ∘ᵥ (μ₂ ◁ F)
+    ; identityˡ = Identity.identityˡ --: actionˡ ∘ᵥ (F ▷ η₁) ∘ᵥ unitorʳ.to ≈ id₂
+    ; identityʳ = Identity.identityʳ --: actionʳ ∘ᵥ (η₂ ◁ F) ∘ᵥ unitorˡ.to ≈ id₂
+    }
+
+
+
+module TensorproductOfHomomorphisms {M₁ M₂ M₃ : Monad 𝒞} {B₂ B'₂ : Bimodule M₂ M₃} {B₁ B'₁ : Bimodule M₁ M₂}
```

Comment:
Ah, now I see the reason for the naming! Hmm, as these files are quite large, I would be tempted to break this up into
`Categories.Bicategory.Construction.Bimodules.Tensorproduct.Bimodules` and ``Categories.Bicategory.Construction.Bimodules.Tensorproduct.Homomorphisms`.

You could re-export some of that in this file.